zilliqa/src/api/zil.rs

```diff
@@ -59,6 +59,35 @@ pub fn rpc_module(node: Arc<Mutex<Node>>) -> RpcModule<Arc<Mutex<Node>>> {
     )
 }
 
+#[derive(Deserialize)]
+#[serde(transparent)]
+struct ZilAddress {
+    #[serde(deserialize_with = "deserialize_zil_address")]
+    inner: Address,
+}
+
+impl From<ZilAddress> for Address {
+    fn from(value: ZilAddress) -> Self {
+        value.inner
+    }
+}
+
+fn deserialize_zil_address<'de, D>(deserializer: D) -> Result<Address, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+
+    let addr = if s.starts_with("zil") {
```

Comment:
Small nit: I'd prefer if the logic went: "Try to parse with `bech32::decode`. If it succeeds, check the HRP is `zil`. If it fails, try to parse as a hex string".

I guess it is equivalent to this, since if the string starts with `zil` then that must be the HRP, but it feels slightly less robust?